packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -306,174 +342,93 @@ class _TreemapState extends State<Treemap> {
-    return Padding(
-      padding: EdgeInsets.all(widget.levelsVisible > 0 ? 1.0 : 0.0),
-      child: Container(
-        decoration: BoxDecoration(border: Border.all(width: 0.5)),
+    if (widget.rootNode == null && widget.nodes.isNotEmpty) {
+      return buildSubTreemaps();
+    } else {
+      return buildTreemapCell();
+    }
+  }
+
+  Widget buildTreemapCell() {
+    if (widget.levelsVisible > 0 && widget.rootNode.children.isNotEmpty) {
+      return Padding(
+        padding: const EdgeInsets.all(1.0),
         child: Column(
           children: [
-            if (widget.levelsVisible > 0 &&
-                rootNode.children.isNotEmpty &&
-                shouldDisplayText)
-              Center(
-                child: Text(
-                  rootNode.name + ' ' + nodeSizeText(),
-                  overflow: TextOverflow.ellipsis,
-                  textAlign: TextAlign.center,
+            // TODO(peterdjlee): Abstract out to a widget or a method.
+            Tooltip(
+              message: widget.rootNode.name,
+              child: InkWell(
+                onTap: () {
+                  widget.onRootChangedCallback(widget.rootNode);
+                },
+                child: Container(
+                  width: double.infinity,
+                  decoration: BoxDecoration(border: Border.all(width: 0.5)),
+                  child: buildNameAndSizeText(
+                    fontColor: Colors.white,
+                    onTwoLines: false,
+                  ),
                 ),
               ),
-            Expanded(child: buildContent()),
+            ),
+            Expanded(
+              child: NewTreemap.fromNodes(
+                nodes: widget.rootNode.children,
+                levelsVisible: widget.levelsVisible - 1,
+                onRootChangedCallback: widget.onRootChangedCallback,
+              ),
+            ),
           ],
         ),
-      ),
-    );
-  }
-
-  Widget buildContent() {
-    if (widget.levelsVisible == 0) {
-      if (shouldDisplayText) {
-        return Container(
-          width: widget.width,
-          child: Center(
-            child: nameAndSizeText(),
-          ),
-        );
-      } else {
-        return const SizedBox();
-      }
-    } else if (widget.levelsVisible == 1) {
-      if (rootNode.children.isNotEmpty) {
-        return GestureDetector(
-          onTap: widget.onTap,
+      );
+    } else {
+      // TODO(peterdjlee): Abstract out to a widget or a method.
+      return Tooltip(
+        message: widget.rootNode.name,
+        child: InkWell(
+          onTap: () {
+            widget.onRootChangedCallback(widget.rootNode);
+          },
           child: Container(
-            color: mainUiColor,
-            child: Tooltip(
-              message: rootNode.name,
-              child: buildNestedTreemap(),
+            decoration: BoxDecoration(
+              color: mainUiColor,
+              border: Border.all(width: 0.5),
```

Comment:
a border width 0.5 will look different on high DPI and regular DPI devices. Is there a reason we can't use border width 1.0 and a color that stands out less?
Not all high dpi devices will have a pixel ratio of 2.0 meaning you will have some subtle differences on high dpi devices as well.

Reply:
That's a good consideration, something I didn't think about. Fixed